### PR TITLE
Replace `@import` with `#import`

### PR DESCRIPTION
**Pull request by ksuther**
Replace `@import` with the equivalent `#import`

### CODE REVIEW

File: custom/ReplaceAtImport.py
Comment (vendruscolo) [inline custom/ReplaceAtImport.py:18]: Can we add a check before this line to ensure the header file exists? I'm wary we might end up (in the future) with a module name with spaces that messes up with underscores and the file system. Especially because we run spacecommander as the last step and we might not compile the codebase afterwards.

File: custom/RemoveAPIAvailableSemicolon.py
Comment (vendruscolo) [inline custom/RemoveAPIAvailableSemicolon.py:1]: Typo, missing `e` in "Smicolon"

**Review by vendruscolo — APPROVED** [review]

File: custom/ReplaceAtImport.py
Comment (alanflex) [reply custom/ReplaceAtImport.py:18]: There are *some* places where this substitution won't work and will need manual intervention, for example: 

https://github.com/flexibits/fantastical/pull/7028/files#diff-9b04d0303eabfb071c21f8fb4a0e255a381d55e817bba6fdb23d7eb27734ee51L13. 

There's also may be some places where we cannot replace the import and Windows will just have to deal... like `@import FantasticalSwift`. 

I'm not sure if it's better to make this a "@import found, please check!" kind of validation rather than try to fix it automatically.  Maybe we need to special case the Swift modules, or look for a magic comment `//necessary-import` or something to avoid them getting caught.

Comment (alanflex) [reply custom/ReplaceAtImport.py:18]: I can't find it now but I was sure I found a line where ObjC was @importing a swift framework and I just left it :-)

File: custom/RemoveAPIAvailableSemicolon.py
Comment (ksuther) [reply custom/RemoveAPIAvailableSemicolon.py:1]: Fixed

File: custom/ReplaceAtImport.py
Comment (ksuther) [reply custom/ReplaceAtImport.py:18]: How would we know if the header exists as a general rule? We could remove this and make it so it isn't an automatic fix, but I'm not sure where we'd put it to make it a warning that's easily bypassed.

Comment (alanflex) [reply custom/ReplaceAtImport.py:18]: I was suggesting that we check for some kind of magic comment after the @import so that would act as a "bypass".  Almost like when you use pragmas to switch off a code warning knowingly.

To be clear, I'm fine with this as long as it's understood that it *may* result in committing code to your local branch that doesn't compile :-). You'd then need to fix up anything weird (like the `ObjectiveC.message` => `<ObjectiveC.message/ObjectiveC.message.h>` => `<objc/message.h>`) before trying to push or merge.

Comment (ksuther) [reply custom/ReplaceAtImport.py:18]: Yeah, even `@import ObjectiveC;` would fail. I'm not sure if failing because of no automatic fix or failing because of an automatic fix is the more likely scenario.

Making it detect something like `// spacecommander-ignore` and skipping would be easy to add, but even if we add it that doesn't prevent accidental import transformation.

Comment (alanflex) [reply custom/ReplaceAtImport.py:18]: Yeah sorry, I was confusing two separate issues.  (1) some imports are not so simple (2) some we may just have to leave because they refer to swift packages and Xcode doesn't care.

Comment (ksuther) [reply custom/ReplaceAtImport.py:18]: I feel like we have two options:

1. Do nothing and wait for `@imports` sneaking in to become a problem.
2. Merge this as it is and see if we dislike the new behavior.

I'm leaning towards 2 because the chances of issues stemming from 2 seems lower, but I don't have a super strong opinion :)

**Comment by alanflex** [comment]
+1 from me.  I just wanted to be alert to the possible compiler errors for occasional weird modules :)Sent from my iPhoneOn 20 Apr 2023, at 17:37, Kent Sutherland ***@***.***> wrote:﻿
@ksuther commented on this pull request.



In custom/ReplaceAtImport.py:
> +#
+# If a filename is specified as a parameter, it will change that file in place.
+# If input is provided through stdin, it will send the result to stdout.
+
+from AbstractCustomFormatter import AbstractCustomFormatter
+import re
+
+class ReplaceAtImport(AbstractCustomFormatter):
+    def format_lines(self, lines):
+
+        lines_to_write = []
+        for line in lines:
+            to_append = line
+
+            if ***@***.*** "):
+                to_append = ***@***.*** (\\w+);", "#import <\\1/\\1.h>", line)

I feel like we have two options:

Do nothing and wait for @imports sneaking in to become a problem.
Merge this as it is and see if we dislike the new behavior.

I'm leaning towards 2 because the chances of issues stemming from 2 seems lower, but I don't have a super strong opinion :)

**Review by alanflex — APPROVED** [review]

**Pull request closed without merging**
